src/Cassandra/Cluster.cs

```diff
@@ -39,7 +39,8 @@ public class Cluster : ICluster
         private readonly ControlConnection _controlConnection;
         private volatile bool _initialized;
         private volatile Exception _initException;
-        private readonly object _initLock = new object();
+        private readonly SemaphoreSlim _initLock = new SemaphoreSlim(0, 1);
```

Comment:
It should be `new SemaphoreSlim(1, 1)` as `initialCount` refers to initially available (its confusing setting!)